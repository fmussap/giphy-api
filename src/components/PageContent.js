import React, { Component } from 'react'

class PageContent extends Component {
  constructor (props) {
    super()
    this.state = {
      src: props.urlStill,
      hover: ''
    }

    this.handleClick = () => {
      props.handleOpenModal(props.urlModal, props.title)
      // this changes the class to disable the hover efect
      this.setState(() => ({
        hover: 'no-hover'
      }))
    }

    // this changes the image to be load from static / animated
    this.handleHoverOver = () => {
      this.setState(() => ({
        src: props.src
      }))
    }

    // it changes the image to be load from animated / static
    this.handleHoverOut = () => {
      this.setState(() => ({
        src: props.urlStill,
        hover: ''
      }))
    }
  }

  render () {
    return (
      <div
        className={`individual-gif-container ${this.state.hover}`}
        style={{ backgroundImage: `url(${this.state.src})`, backgroundRepeat: 'no-repeat' }}
        onClick={this.handleClick}
        onMouseOver={this.handleHoverOver}
        onMouseOut={this.handleHoverOut}
      >
        <div className='individual-gif-info'>{this.props.title}</div>
      </div>
    )
  }
}

export default PageContent
