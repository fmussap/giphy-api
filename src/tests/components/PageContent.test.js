import React from 'react'
import { shallow } from 'enzyme'
import PageContent from '../../components/PageContent'

let spy, wrapper
beforeEach(() => {
  spy = jest.fn()
  const data = {
    urlStill: 'http:/test/still',
    urlModal: 'http:/test',
    title: 'Any title',
    src: 'http:/test/src',
    handleOpenModal: spy
  }
  wrapper = shallow(<PageContent
    {...data}
  />)
})

test('should render Page correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle onClick', () => {
  wrapper.find('div.individual-gif-container').simulate('click')
  expect(spy).toHaveBeenLastCalledWith('http:/test', 'Any title')
  expect(wrapper.state().hover).toBe('no-hover')
})

test('should handle onMouseOver', () => {
  wrapper.find('div.individual-gif-container').simulate('mouseOver')
  expect(wrapper.state().src).toBe('http:/test/src')
})

test('should handle onMouseOut', () => {
  wrapper.find('div.individual-gif-container').simulate('mouseOut')
  expect(wrapper.state().src).toBe('http:/test/still')
})
