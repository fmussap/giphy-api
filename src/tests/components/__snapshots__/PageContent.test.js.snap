// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Page correctly 1`] = `
<div
  className="individual-gif-container "
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  style={
    Object {
      "backgroundImage": "url(http:/test/still)",
      "backgroundRepeat": "no-repeat",
    }
  }
>
  <div
    className="individual-gif-info"
  >
    Any title
  </div>
</div>
`;
